src/game/server/gamecontroller.h

```diff
@@ -150,6 +151,7 @@ class IGameController
 
 	CClientMask GetMaskForPlayerWorldEvent(int Asker, int ExceptID = -1);
 	virtual void InitTeleporter();
+	virtual CServerInfo::EClientScoreKind ScoreKind(int ClientID) const { return CServerInfo::CLIENT_SCORE_KIND_TIME; };
```

Comment:
```suggestion
	virtual CServerInfo::EClientScoreKind ScoreKind() const { return CServerInfo::CLIENT_SCORE_KIND_TIME; };
```
Shouldn't support making it dependent on the snapping client, it's basically unsupported by the protocol. There can only be one score kind in the server info, and only one score kind for all of the scoreboard for each client, so it doesn't make sense to mix them.

Irrelevant, see top-level review.

Reply:
This is also called in the snap where it does actually work to show timescore in the scoreboard for one player and point score for another.

Reply:
> > and only one score kind for all of the scoreboard for each client, so it doesn't make sense to mix them.
>
> This is also called in the snap where it does actually work to show timescore in the scoreboard for one player and point score for another.

Yes, but only one score kind for the whole scoreboard in one client.

Reply:
Yes. ``ClientID`` is the SnappingClient

Reply:
```C++
void IGameController::Snap(int SnappingClient)
{
	// [..]

	CNetObj_GameInfoEx *pGameInfoEx = Server()->SnapNewItem<CNetObj_GameInfoEx>(0);
	if(!pGameInfoEx)
		return;

	pGameInfoEx->m_Flags =
		GAMEINFOFLAG_GAMETYPE_RACE |
		GAMEINFOFLAG_GAMETYPE_DDRACE |
		GAMEINFOFLAG_GAMETYPE_DDNET |
		GAMEINFOFLAG_UNLIMITED_AMMO |
		GAMEINFOFLAG_RACE_RECORD_MESSAGE |
		GAMEINFOFLAG_ALLOW_EYE_WHEEL |
		GAMEINFOFLAG_ALLOW_HOOK_COLL |
		GAMEINFOFLAG_ALLOW_ZOOM |
		GAMEINFOFLAG_BUG_DDRACE_GHOST |
		GAMEINFOFLAG_BUG_DDRACE_INPUT |
		GAMEINFOFLAG_PREDICT_DDRACE |
		GAMEINFOFLAG_PREDICT_DDRACE_TILES |
		GAMEINFOFLAG_ENTITIES_DDNET |
		GAMEINFOFLAG_ENTITIES_DDRACE |
		GAMEINFOFLAG_ENTITIES_RACE |
		GAMEINFOFLAG_RACE;
	if(ScoreKind(SnappingClient) == CServerInfo::CLIENT_SCORE_KIND_TIME)
		pGameInfoEx->m_Flags |= GAMEINFOFLAG_TIMESCORE;
```

The way I understood it this is called once for every client. And ``SnappingClient`` is the recipient of that snap. So for all of them it might be a different score kind if the server has mini games where one half of the player base sees score kinds and the other sees race times.

Reply:
> Yes. `ClientID` is the SnappingClient

Yes.

I'm saying:
1) This should not be supported by the source code, because it can only show the complete scoreboard in a different view, for a particular client.
2) This is irrelevant, because I want to remove this method completely, replacing it by a method appending to the JSON in mod controller (passed through from the engine).
[3) If this method were to stay, `ClientID` would need to be renamed to `SnappingClientID` or similar.]

Reply:
2. This is not about the master server or json. This is about the snap.
3. done

Reply:
Okay I removed the argument -.-
Now modders who need a per client score kind have to overwrite and reimplement the whole snap method.